### DISCUSSION TITLE
Appending a list

### DISCUSSION THREAD

**Discussion by bocc** [mitsuhiko/minijinja#414] [Q&A] [answered]
Hello, apologies for the basic question, but I have found no mention of this in the documentation, and I'm not really deep into either Jinja2 or minijinja. Looked in the contrib crate as well, but no luck. My goal is to append a list:

```
{% set some_list = [0,1,2] %}
{% do some_list.append(3) %}
```
But the `list` object has no `append` method. How should I approach this?

After some looking into, I see that `SeqObject` has no `add`/`insert` or similar methods (as slices need to implement it). The workaround I came up with is to define a function that implements this, but I still hope there is a better option.

**Comment by mitsuhiko** [comment, marked as answer]
They are immutable. You can only add an append filter and then to something like this:

```
{% set some_list = [0, 1, 2] %}
{% set some_list = some_list|append(3) %}
```

**Reply by bocc** [reply]
@mitsuhiko thanks for your reply! Do you think implementing this makes sense, would you consider a PR with this? It seems like a step towards feature parity with other Jinja2 implementations.

**Reply by mitsuhiko** [reply]
The demand for this seems relatively low and `|append` in this form does not exist in other engines either. I'm not sure what the right course of action is. I would play around it in an external crate first.

**Reply by fortzi** [reply]
While I personally like the decision not to generally support python methods, I do think that implementing some select ones can be immensely helpful in migrating existing projects from other implementations.
Some notable examples are `List.append`, `List.extend`, `Dict.update`, `datetime.strftime`, `datetime.isoformat`, etc.
While it's possible to implement filters to replace such methods, migrating every template that uses them is the real obstacle.
WDYT?

**Reply by fortzi** [reply]
It's worth adding that `{% set some_list = some_list|append(3) %}` doesn't work inside a loop 🤷

**Reply by mitsuhiko** [reply]
The python methods that are not modifying values are implemented with `pycompat`.  Methods that change collections are tricker since the objects themselves are not synchronized for the most part.

`{% set ns.some_list = ns.some_list|append(3) %}` does work in a loop if you really need to.
